lualib/rspamadm/mime.lua

```diff
@@ -873,19 +879,85 @@ local function sign_handler(opts)
   end
 end
 
+-- Strips directories and .extensions (if present) from a filepath
+  -- very_simple
+  -- /home/very_simple.eml
+  -- very_simple.eml
+  -- very_simple.example.eml
+  -- /home/very_simple
+  -- home/very_simple
+  -- ./home/very_simple
+  -- ../home/very_simple.eml
+--All the above end up as very_simple
+local function filename_only(filepath)
```

Comment:
What if there are more than one extension, e.g. `tar.gz`? Currently, there is a code that does it in the following way: https://github.com/rspamd/rspamd/blob/master/src/plugins/lua/mime_types.lua#L210
But it will likely be broken with a full path: `dir.with.dot/ololo` will not be processed correctly. However, I'm not sure that your function will process it correctly either.

Reply:
Multiple file extensions are handled as per [line 886](https://github.com/JasonStephenson/rspamd/blob/6460f4088b8f8fcaa167ee0cef3505d7ded9d6ec/lualib/rspamadm/mime.lua#L886)

My function correctly handles directory names with dots
`print(filename_only("/home/a.directory/very_simple.eml")) --> very_simple`

Reply:
Could you please add some asserts like that directly in the code, so we can ensure that it is not broken by further changes. It will slightly slow down the loading of the file but it is negligible comparing to the benefits from the correctness. You can check the following example to get an idea: https://github.com/rspamd/rspamd/blob/master/lualib/lua_aws.lua#L281

Reply:
done